pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -321,38 +322,40 @@ func (e *Controller) processNextWorkItem(ctx context.Context) bool {
 	}
 	defer e.queue.Done(eKey)
 
+	logger := klog.FromContext(ctx)
 	err := e.syncService(ctx, eKey.(string))
-	e.handleErr(err, eKey)
+	e.handleErr(logger, err, eKey)
 
 	return true
 }
 
-func (e *Controller) handleErr(err error, key interface{}) {
+func (e *Controller) handleErr(logger klog.Logger, err error, key interface{}) {
 	if err == nil {
 		e.queue.Forget(key)
 		return
 	}
 
 	ns, name, keyErr := cache.SplitMetaNamespaceKey(key.(string))
 	if keyErr != nil {
-		klog.ErrorS(err, "Failed to split meta namespace cache key", "key", key)
+		logger.Error(err, "Failed to split meta namespace cache key", "key", key)
 	}
 
 	if e.queue.NumRequeues(key) < maxRetries {
-		klog.V(2).InfoS("Error syncing endpoints, retrying", "service", klog.KRef(ns, name), "err", err)
+		logger.V(2).Info("Error syncing endpoints, retrying", "service", klog.KRef(ns, name), "err", err)
 		e.queue.AddRateLimited(key)
 		return
 	}
 
-	klog.Warningf("Dropping service %q out of the queue: %v", key, err)
+	logger.Info("Dropping service out of the queue", "service", klog.KRef(ns, name), "err", err)
 	e.queue.Forget(key)
 	utilruntime.HandleError(err)
 }
 
 func (e *Controller) syncService(ctx context.Context, key string) error {
 	startTime := time.Now()
+	logger := klog.FromContext(ctx)
 	defer func() {
-		klog.V(4).Infof("Finished syncing service %q endpoints. (%v)", key, time.Since(startTime))
+		logger.V(4).Info("Finished syncing service endpoints", "service", key, "startTime", time.Since(startTime))
```

Comment:
/lgtm cancel

I missed this initially, but then saw it in another PR: `service` is used inconsistently. Sometimes it's the work queue key (a string), sometimes the object (from `klog.KObj`).

Let's use `key` for the string and `service` for the object. 

This also applies in some other places, please search/replace.

Reply:
Updated, thanks